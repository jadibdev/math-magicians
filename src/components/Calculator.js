import React, { useState } from 'react';
import Screen from './Screen';
import calculate from '../logic/calculate';
import Header from './Header';

const Calculator = () => {
  const [State, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (e) => {
    const display = () => calculate(State, e.target.innerHTML);
    setState(display);
  };

  let result = 0;
  const { total, next, operation } = State;
  if (next === null && operation === null && total === null) {
    result = 0;
  }

  if (next !== null && total === null) {
    result = next;
  }

  if (next === null && operation !== null && total !== null) {
    result = total + operation;
  }

  if (next !== null && operation !== null && total !== null) {
    result = total + operation + next;
  }

  if (next === null && operation === null && total !== null) {
    result = total;
  }

  return (
    <div>
      <Header />
      <h2 id="calculator-cta">Let&apos;s do some Math:</h2>
      <div className="container">
        <Screen result={result} />
        <button type="button" className="item" onClick={handleClick}>
          AC
        </button>
        <button type="button" className="item" onClick={handleClick}>
          +/-
        </button>
        <button type="button" className="item" onClick={handleClick}>
          %
        </button>
        <button type="button" className="item orange" onClick={handleClick}>
          รท
        </button>
        <button type="button" className="item" onClick={handleClick}>
          7
        </button>
        <button type="button" className="item" onClick={handleClick}>
          8
        </button>
        <button type="button" className="item" onClick={handleClick}>
          9
        </button>
        <button type="button" className="item orange" onClick={handleClick}>
          x
        </button>
        <button type="button" className="item" onClick={handleClick}>
          4
        </button>
        <button type="button" className="item" onClick={handleClick}>
          5
        </button>
        <button type="button" className="item" onClick={handleClick}>
          6
        </button>
        <button type="button" className="item orange" onClick={handleClick}>
          +
        </button>
        <button type="button" className="item" onClick={handleClick}>
          3
        </button>
        <button type="button" className="item" onClick={handleClick}>
          2
        </button>
        <button type="button" className="item" onClick={handleClick}>
          1
        </button>
        <button type="button" className="item orange" onClick={handleClick}>
          -
        </button>
        <button type="button" className="item zero" onClick={handleClick}>
          0
        </button>
        <button type="button" className="item dot" onClick={handleClick}>
          .
        </button>
        <button type="button" className="item equals orange" onClick={handleClick}>
          =
        </button>
      </div>
    </div>
  );
};

export default Calculator;
