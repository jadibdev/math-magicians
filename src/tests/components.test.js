import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';
import Home from '../components/Home';

it('Calculator component renders correctly', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Home component renders correctly', () => {
  const tree = renderer
    .create(<Home />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Quote component renders correctly', () => {
  const tree = renderer
    .create(<Home />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
