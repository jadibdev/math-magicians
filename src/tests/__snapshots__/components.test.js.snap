// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component renders correctly 1`] = `
<div>
  <div
    className="navContainer"
  >
    <div>
      <h1>
        Math Magicians
      </h1>
    </div>
    <div>
      <ul
        className="nav"
      >
        <li>
          Home
        </li>
        <li>
          Calculator
        </li>
        <li>
          Quote
        </li>
      </ul>
    </div>
  </div>
  <h2
    id="calculator-cta"
  >
    Let's do some Math:
  </h2>
  <div
    className="container"
  >
    <div
      className="item screen"
    >
      <p>
        0
      </p>
    </div>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="item orange"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="item orange"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="item orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="item orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="item zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="item dot"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="item equals orange"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Home component renders correctly 1`] = `
<div>
  <div
    className="navContainer"
  >
    <div>
      <h1>
        Math Magicians
      </h1>
    </div>
    <div>
      <ul
        className="nav"
      >
        <li>
          Home
        </li>
        <li>
          Calculator
        </li>
        <li>
          Quote
        </li>
      </ul>
    </div>
  </div>
  <div
    className="home-container"
  >
    <h1>
      Welcome to our page!
    </h1>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, tenetur error, harum nesciunt ipsum debitis quas aliquid. Reprehenderit, quia. Quo neque error repudiandae fuga? Ipsa laudantium molestias eos sapiente officiis modi at sunt excepturi expedita sint? Sed quibusdam recusandae alias error harum maxime adipisci amet laborum. Perspiciatis minima nesciunt dolorem! Officiis iure rerum voluptates a cumque velit quibusdam sed amet tempora. Sit laborum ab, eius fugit doloribus tenetur fugiat, temporibus enim commodi iusto libero magni deleniti quod quam consequuntur! Commodi minima excepturi repudiandae velit hic maxime
    </p>
  </div>
</div>
`;

exports[`Quote component renders correctly 1`] = `
<div>
  <div
    className="navContainer"
  >
    <div>
      <h1>
        Math Magicians
      </h1>
    </div>
    <div>
      <ul
        className="nav"
      >
        <li>
          Home
        </li>
        <li>
          Calculator
        </li>
        <li>
          Quote
        </li>
      </ul>
    </div>
  </div>
  <div
    className="home-container"
  >
    <h1>
      Welcome to our page!
    </h1>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, tenetur error, harum nesciunt ipsum debitis quas aliquid. Reprehenderit, quia. Quo neque error repudiandae fuga? Ipsa laudantium molestias eos sapiente officiis modi at sunt excepturi expedita sint? Sed quibusdam recusandae alias error harum maxime adipisci amet laborum. Perspiciatis minima nesciunt dolorem! Officiis iure rerum voluptates a cumque velit quibusdam sed amet tempora. Sit laborum ab, eius fugit doloribus tenetur fugiat, temporibus enim commodi iusto libero magni deleniti quod quam consequuntur! Commodi minima excepturi repudiandae velit hic maxime
    </p>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div>
  <div
    className="navContainer"
  >
    <div>
      <h1>
        Math Magicians
      </h1>
    </div>
    <div>
      <ul
        className="nav"
      >
        <li>
          Home
        </li>
        <li>
          Calculator
        </li>
        <li>
          Quote
        </li>
      </ul>
    </div>
  </div>
  <h2
    id="calculator-cta"
  >
    Let's do some Math:
  </h2>
  <div
    className="container"
  >
    <div
      className="item screen"
    >
      <p>
        0
      </p>
    </div>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="item orange"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="item orange"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="item orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="item"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="item orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="item zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="item dot"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="item equals orange"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
